rust/bitcoin_wallet/canister/README.adoc

```diff
@@ -0,0 +1,149 @@
+= The Bitcoin Canister
```

Comment:
Why do we need the Bitcoin integration API here?
I think we only need  one top-level README explaining the functionality of the wallet and with links to additional documentation.

Reply:
I guess by API you mean `README.md`. Otherwise without `canister/` we can't in a standalone way test the Bitcoin wallet.

I will remove the `canister/README.md` and complete `README.md`

Reply:
More general question: Why is the entire canister code copied? If we need it (right now), we should import it as a dependency in order to prevent any divergence.

Reply:
Thank you for shedding some light on this point.
In fact it's not an easy `Cargo.toml` line as `ic-bitcoin-canister = "0.X"` line to add.

Current recipe is:

```
"btc": {
  "type": "custom",
  "candid": "canister/candid.did",
  "wasm": "target/wasm32-unknown-unknown/release/canister.wasm",
  "build": "scripts/build-canister.sh"
},
```

I would do something similar to the Internet Identity recipe:

```
"btc": {
  "type": "custom",
  "candid": "bitcoin_canister.did",
  "wasm": "bitcoin_canister.wasm",
  "build": "curl -sSL https://github.com/USER/REPOSITORY/releases/latest/download/bitcoin_canister.wasm -o bitcoin_canister.wasm"
},
````

The question is then do we make a release of the Bitcoin canister in `dfinity/bitcoin-developer-preview`, `dfinity/bitcoin-canister` or something more temporary like `BenjaminLoison/bitcoin-canister` ?

Reply:
You are right, you can't add it in the form of a crate but you should be able to add it as follows:

``ic-btc-canister = { git = "https://github.com/dfinity/ic.git", rev = "[some commit hash]" }``

Reply:
In fact when I say that we need "`canister/` [to test] in a standalone way [...] the Bitcoin wallet."
I mean that it's not a library dependency (so the discussion isn't about `Cargo.toml` but `dfx.json` as far as I know). At the end of the compilation we need a wam to pass to dfx for executing a Bitcoin canister in a standalone way otherwise these commandes from the `README.md` instructions wouldn't succeed as far as I know:

```
$ dfx deploy btc
$ cargo run --features="tokio candid ic-agent garcon tonic tonic-build" --bin adapter-shim $(dfx canister id btc)

$ dfx deploy bitcoin_wallet_assets --argument "(record { bitcoin_canister_id = principal \"$(dfx canister id btc)\" })"
```

Am I wrong ?

Reply:
In fact it's not even possible to do it exactly the way I suggested because:

```
cargo run --features="tokio candid ic-agent garcon tonic tonic-build" --bin adapter-shim $(dfx canister id btc)
```

requires to have `canister/src/adapter-shim.rs` so I would say we should remove `canister/` and update the `README.md` to:

## Usage

The following commands will start a replica, install the Bitcoin canister from the bitcoin-developer-preview and the development Internet Identity canister, and deploy the Bitcoin wallet webapp:

```bash
$ dfx start --background --clean
$ git clone https://github.com/dfinity/bitcoin-developer-preview.git
$ cd bitcoin-developer-preview/
$ dfx deploy btc
$ cargo run --features="tokio candid ic-agent garcon tonic tonic-build" --bin adapter-shim $(dfx canister id btc)

$ npm install
$ dfx deploy internet_identity --argument '(null)'
$ dfx deploy bitcoin_wallet_assets --argument "(record { bitcoin_canister_id = principal \"$(dfx canister id btc)\" })"
```

What is your opinion on this ?

Reply:
Yes, you are right. Obviously, we won't need it later because we will interact with the proper Bitcoin API but we need it now. So, let's keep it in for now.